### DISCUSSION TITLE
Not all notifications send through all channels / to all accounts?

### DISCUSSION THREAD

**Discussion by Starfiresg1** [alexjustesen/speedtest-tracker#1213] [Q&A]
Hi,

I just upgraded to v16.6 from v15 and noticed that the notifications regarding the data migration (notification about the change, start and finish of the migration) where only displayed in the GUI top right.
I have set up webhook notifications (test message works) but these notifications didn't show up in the webhook endpoint?
Is this supposed to be like this?

Furthermore I have setup a second account with a different e-mail than admin@example.com which too has the admin role.
When logged with this account I got the database migration notifications in the top right.
To be save I decided to do the migration with the default admin@exmaple.com account.
There I noticed that I had a lot of notifications that didn't appear in the second account.

The admin@example.com account had threshold messages that newer appeared on the second account. Didn't miss them because I didn't notice the treshold being breached.

Threshold and notification settings
![grafik](https://github.com/alexjustesen/speedtest-tracker/assets/8659009/03bb17ec-b40f-4b16-a686-7e49c8eb8f0e)
![grafik](https://github.com/alexjustesen/speedtest-tracker/assets/8659009/ee41af20-cc62-46bc-a99c-798e4f647d0b)

Strangely the threshold notifications only appear in the admin@example.com account - not in the second account.

Seems to me all notifications are currently only send to the first user? While the webhook test works the treshold messages aren't sent to the endpoint at all.

Edited: Took a look at the source code - seems to be not compatible with my endpoint (Gotify):
The test notification sends a "message" as payload - which is supported and required by Gotify (additionally priority and title are supported). So the test message works fine.
On speedtest run or threshold failure the payload doesn't contain a message at all - instead 'result_id', 'site_name' and 'metrics' in case of threshold failure. So the messages are probably discarded by Gotify because the 'message' is required and missing. From Gotify it probably considers the notification "empty".

**Comment by alexjustesen** [comment]
The quick answers:

"Notifications" section of the panel only applies to speedtests so when you run maintenance or other features of the application and a UI notification is shown it's only being sent to the user who triggered it. Additionally the database notification regarding the data migration that are shown in the UI sent to all `admin` users is intended to force and communicate that action needs to be taken. This way I don't have to assume which admin account people are using in case they have multiple.

#40 also Gotify is not currently supported.

**Reply by Starfiresg1** [reply]
Thanks for clarifying. I noticed the Gotify part when I looked at the source - I got tripped up because the test message is working because it's compatible (probably by chance).

Could you clarify that it is expected that database messages regarding threshold misses are only sent to default admin account (ID 1) and not to any other (admin) accounts that are set up? Because I noticed this happening.
